import_types_temp/module_type_extractor.py

```diff
@@ -0,0 +1,353 @@
+        # and then we concatenate each of the import statement lists together to finally form
+        # a coherent list of import statements.
+
+        # Helper function to combine two paths (or one list of ImportStatements and one path)
+        # to a coherent list of ImportStatements
+        def reduce_paths(p1, p2):
+            p2_statements = self.get_file_import_statements(p2)
+
+            # First path is a string. That means this is the first reduction operation,
+            # and we must first convert the path to a list of import statements.
+            if (isinstance(p1, str)):
+                # Combine path1 import statements & path 2 import statements
+                return self.get_file_import_statements(p1) + p2_statements
+            else:
+                # Combine previous import statements with path 2 import statements
+                return p1 + p2_statements
+
+
+        import_statements = reduce(reduce_paths, import_statements)
+
+        return import_statements
+
+
+
+    def get_module(self, import_name):
+        """
+        Resolves the specified import to a module/package, or returns None
+        if that is not possible.
+
+        Example import name: scikit.linalg
+
+        :param: import_name  Name of the import (full import name) to resolve
+        :return: Module object or None
+        """
+        # Old code to load module from system
+        # Module not in system; We load the module.
+        # if (name not in sys.modules):
+        #     spec = importlib.util.spec_from_file_location(name, path)
+        #     module = importlib.util.module_from_spec(spec)
+        #     spec.loader.exec_module(module)
+        # else:
+        #     # Module in system; Retrieve directly
+        #     module = sys.modules[name]
+
+        # Check whether the import entry is a module. It is possible that the
+        # import statement refers to an import of a module member (e.g. function or class),
+        # in which case the import will be resolved to 'None'. Otherwise, the import
+        # is resolved to a non-None path. Note that the path will be resolved correctly
+        # if the import is a star statement (i.e. it will return a non-None path)
+        is_module = import_finder._resolve_import(import_name) is not None
+        
+        # Import is a module if it has a path
+        if (is_module):
+            # Retrieve the package given by the import name. We have to split by the '.'
+            # to conform to the function call's signature.
+            # Note that the package will be found, since the _resolve_import method above
+            # will load the module as well, and if a path is returned, we will get the package.
+            i, module = import_finder._find_package(import_name.split("."))
+            return module
+        else:
+            return None
+
+    def is_from_import_type(self, import_name, import_from):
```

Comment:
TODO: The method signature & description is outdated. This should be updated to reflect that it actually now returns the type.